### ISSUE TITLE
Bad translation of identity to CockroachDB

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10423]
CockroachDB doesn't support identity columns yet, only PostgreSQL's historic serial types. However, this isn't being translated correctly:

```sql
create table i (i int not null auto_increment primary key)
```

Output:

```sql
create table I (
  I int4 not null,
  primary key (I)
)
```

**Comment by lukaseder** [comment]
The status quo doesn't produce any `DEFAULT` expression at all, so this is definitely not good. But an emulation seems hard as well as the default of using `unique_rowid()` might produce collisions, if we truncate it or modulo it with `2 ^ 31`:

```sql
select count(*) filter (where c > 1)
from (
  select v, count(*) c
  from (
    select unique_rowid() % 2 ^ 31 v
    from generate_series(1, 100000)
  ) t
  group by v
  having count(*) > 1
) t;
```

Produces: `34464`. I've checked in the above to fix some integration tests, but clearly, we need to do better.

**Comment by lukaseder** [comment]
Well, even for `INT8`, CRDB can't guarantee uniqueness, it seems:

>  This behavior is statistically likely to be globally unique except in extreme cases (see this FAQ entry for more details).

From https://www.cockroachlabs.com/docs/stable/serial.html

Perhaps that's expected and users are more likely to use UUIDs?

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.16.0. I won't backport to 3.15, given that the fix may not be complete for `INT4`
